src/main/java/nextstep/subway/domain/Distance.java

```diff
@@ -0,0 +1,34 @@
+package nextstep.subway.domain;
+
+import javax.persistence.Column;
+import javax.persistence.Embeddable;
+
+@Embeddable
+public class Distance {
+    @Column(name = "distance")
+    private Long distance;
+
+    protected Distance() {}
+
+    private Distance(Long distance) {
+        this.distance = distance;
+    }
+
+    public static Distance from(Long distance) {
+        if (distance <= 0) {
+            throw new IllegalArgumentException("역 사이의 거리는 0이하일 수 없습니다.");
+        }
+        return new Distance(distance);
+    }
+
+    public void subtract(Distance distance) {
+        if (this.distance <= distance.distance) {
+            throw new IllegalArgumentException("새로운 역이 기존 역 사이보다 크거나 같으면 등록할 수 없다");
+        }
+        this.distance -= distance.distance;
```

Comment:
이 부분은 제가 바로 전 리뷰에서 놓친 부분입니다만...ㅎ.ㅎ;;; VO는 [보통 불변객체로 만들어줍니다](https://tecoble.techcourse.co.kr/post/2020-06-11-value-object/).
그래서 이 부분도 이렇게 변경해줄 수 있을 것 같아요...!
```suggestion
        return Distance.from(this.distance - distance.distance)
```